pkg/xds/server/callbacks/dataplane_metadata_tracker.go

```diff
@@ -47,6 +49,9 @@ func (d *DataplaneMetadataTracker) OnStreamRequest(stream int64, req util_xds.Di
 		// This holds true regardless of the acceptance of the discovery responses on the same stream.
 		// The node identifier should always be identical if present more than once on the stream.
 		// It is sufficient to only check the first message for the node identifier as a result.
+		if d.Metadata(stream) == nil {
+			return errors.New("Metadata is empty for stream")
+		}
```

Comment:
So IIUC, this would be the second request on the xDS stream and we would have already found that there was no metadata on the previous request.

1. Why is shutting down the xDS stream the right choice here?
1. Why don't we detect the error on the first request when we presumably see a node identifier with no metadata?